### PR TITLE
refactor(google-cloud-logging): unify google-cloud-oauth.lua file

### PR DESCRIPTION
**Pull request by HuanXin-Chen**
### Description
Issue: [unify `google-cloud-logging/oauth.lua` and `utils/google-cloud-oauth.lua` file · Issue #11576 · apache/apisix (github.com)](https://github.com/apache/apisix/issues/11576)

What I do :
● scopes -> scope : according to OAuth2/OIDC rules, scope should not be used in the plural
● using utils/google-cloud-oauth.lua replace google-cloud-logging/oauth.lua

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/plugins/google-cloud-logging.lua [hunk 1]
```diff
@@ -43,7 +43,7 @@ local schema = {
                     default = "https://oauth2.googleapis.com/token"
                 },
                 -- https://developers.google.com/identity/protocols/oauth2/scopes#logging
-                scopes = {
+                scope = {
```
Comment (shreemaan-abhishek) [inline apisix/plugins/google-cloud-logging.lua:46]: why should this be renamed?

Comment (HuanXin-Chen) [reply apisix/plugins/google-cloud-logging.lua:46]: According to OAuth2/OIDC rules, scope should not be used in the plural.
image

**Review by shreemaan-abhishek — DISMISSED** [review]

**Comment by HuanXin-Chen** [comment]
@shreemaan-abhishek  Sorry, I forgot to update the document before. Please reiview it again when you have time.Thanks.

**Review by shreemaan-abhishek — APPROVED** [review]

**Review by moonming — DISMISSED** [review]

File: apisix/plugins/google-cloud-logging.lua (hunk 1 above)
Comment (bzp2010) [reply apisix/plugins/google-cloud-logging.lua:46]: @HuanXin-Chen 

So this means destructive updates right?

I'd suggest keeping `scopes` as an alias for `scope` for now, and documenting that it's deprecated. (That is, if `scopes` is used in the configuration, our code needs to rewrite it as `scope` to be passed to google_oauth)

That way users won't experience breakage issues when upgrading and the documentation will remind them to change this field.

Comment (HuanXin-Chen) [reply apisix/plugins/google-cloud-logging.lua:46]: Good point, I'll make the changes tonight. Thank you, mentor.🫡

**Comment by HuanXin-Chen** [comment]
@bzp2010 @moonming @shreemaan-abhishek 

[About Discussion](https://github.com/apache/apisix/pull/11596#discussion_r1783787817)

I have made some new changes: I added a scopes field without a default value to maintain compatibility with existing apisix users.

When the user originally configured the scopes, the scopes will be assigned to the scope, ensuring compatibility. If the user did not configure the scopes, they will use the default value of scope or the scope value configured by the user.

Please reiview it again when you have time.Thanks.

File: apisix/plugins/google-cloud-logging.lua (hunk 1 above)
Comment (HuanXin-Chen) [reply apisix/plugins/google-cloud-logging.lua:46]: Resolved.

**Review by bzp2010 — APPROVED** [review]

**Review by shreemaan-abhishek — APPROVED** [review]

**Review by moonming — DISMISSED** [review]

**Review by bzp2010 — APPROVED** [review]

**Comment by HuanXin-Chen** [comment]
All fixed🫡

**Review by shreemaan-abhishek — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]
LGTM

**Pull request merged**